From lmb@cloudflare.com  Mon Oct 25 20:46:09 2021
From: Lorenz Bauer <lmb@cloudflare.com>
Date: Mon, 25 Oct 2021 17:06:19 +0100
Subject: selftests: bpf: fix backported ASSERT_FALSE
To: gregkh@linuxfoundation.org, andrii@kernel.org
Cc: stable@vger.kernel.org, Lorenz Bauer <lmb@cloudflare.com>
Message-ID: <20211025160619.501058-1-lmb@cloudflare.com>

From: Lorenz Bauer <lmb@cloudflare.com>

Commit 183d9ebd449c ("selftests/bpf: Fix core_reloc test runner") causes
builds of selftests/bpf to fail on 5.10.y since the branch doesn't have the
ASSERT_FALSE macro yet. Replace ASSERT_FALSE with ASSERT_EQ.

Fixes: 183d9ebd449c ("selftests/bpf: Fix core_reloc test runner")
Signed-off-by: Lorenz Bauer <lmb@cloudflare.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 tools/testing/selftests/bpf/prog_tests/core_reloc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tools/testing/selftests/bpf/prog_tests/core_reloc.c
+++ b/tools/testing/selftests/bpf/prog_tests/core_reloc.c
@@ -857,7 +857,7 @@ void test_core_reloc(void)
 			goto cleanup;
 		}
 
-		if (!ASSERT_FALSE(test_case->fails, "obj_load_should_fail"))
+		if (!ASSERT_EQ(test_case->fails, false, "obj_load_should_fail"))
 			goto cleanup;
 
 		equal = memcmp(data->out, test_case->output,
